src/UI/FSMPMenu.cpp

```
@@ -0,0 +1,1582 @@
+	};
+
+	std::vector<PresetEntry> g_presets;
+	bool g_presetsScanned = false;
+
+	// Read every *.json under configsPresets/ and parse it once into a cached struct, so the per-frame
+	// "is this the active preset?" comparison is a cheap struct compare rather than re-reading files.
+	void scanPresets()
+	{
+		g_presets.clear();
+		namespace fs = std::filesystem;
+		std::error_code ec;
```

Clang (14.0.6)

[warning] variable name 'ec' is too short, expected at least 3 characters

(readability-identifier-length)